From dec93a8cf9393268a5b0c10b38479e59bd241b37 Mon Sep 17 00:00:00 2001
From: Ernestas Kulik <ekulik@redhat.com>
Date: Mon, 27 May 2019 16:20:51 +0200
Subject: [PATCH 4/4] general: Drop in-tree copy of gnome-desktop
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The code was copied to avoid having an external GTK+ 3 dependency, but
at this point it’s more trouble than it’s worth.
---
 build-aux/flatpak/org.gnome.Nautilus.json     |   15 +
 build-aux/flatpak/org.gnome.Nautilus.yml      |   10 +
 .../flatpak/org.gnome.NautilusMaster.yml      |   10 +
 meson.build                                   |   50 +-
 .../gnome-desktop-thumbnail-script.c          |  882 -----------
 .../gnome-desktop-thumbnail-script.h          |   38 -
 src/gnome-desktop/gnome-desktop-thumbnail.c   | 1303 -----------------
 src/gnome-desktop/gnome-desktop-thumbnail.h   |  102 --
 src/gnome-desktop/gnome-desktop-update.sh     |   45 -
 src/meson.build                               |    7 +-
 src/nautilus-properties-window.c              |    2 +-
 src/nautilus-thumbnails.c                     |    2 +-
 12 files changed, 39 insertions(+), 2427 deletions(-)
 delete mode 100644 src/gnome-desktop/gnome-desktop-thumbnail-script.c
 delete mode 100644 src/gnome-desktop/gnome-desktop-thumbnail-script.h
 delete mode 100644 src/gnome-desktop/gnome-desktop-thumbnail.c
 delete mode 100644 src/gnome-desktop/gnome-desktop-thumbnail.h
 delete mode 100755 src/gnome-desktop/gnome-desktop-update.sh

diff --git a/build-aux/flatpak/org.gnome.Nautilus.json b/build-aux/flatpak/org.gnome.Nautilus.json
index 9285c4e50..ac1ee9c8d 100644
--- a/build-aux/flatpak/org.gnome.Nautilus.json
+++ b/build-aux/flatpak/org.gnome.Nautilus.json
@@ -120,6 +120,21 @@
         }
       ]
     },
+    {
+        "name": "gnome-desktop",
+        "buildsystem": "meson",
+        "build-options": [
+          "config-opts": [
+            "-Ddebug_tools=false",
+            "-Ddesktop_docs=false",
+            "-Dudev=disabled"
+          ]
+        ],
+        "sources": [
+          "type": "git",
+          "url": "https://gitlab.gnome.org/GNOME/gnome-desktop.git"
+        ]
+    },
     {
       "name": "gnome-autoar",
       "sources": [
diff --git a/build-aux/flatpak/org.gnome.Nautilus.yml b/build-aux/flatpak/org.gnome.Nautilus.yml
index 186746412..1799e9df3 100644
--- a/build-aux/flatpak/org.gnome.Nautilus.yml
+++ b/build-aux/flatpak/org.gnome.Nautilus.yml
@@ -92,6 +92,16 @@ modules:
   - type: git
     url: https://gitlab.gnome.org/GNOME/tracker-miners.git
     tag: 2.2.1
+- name: gnome-desktop
+  buildsystem: meson
+  build-options:
+    config-opts:
+    - "-Ddebug_tools=false"
+    - "-Ddesktop_docs=false"
+    - "-Dudev=disabled"
+  sources:
+  - type: git
+    url: https://gitlab.gnome.org/GNOME/gnome-desktop.git
 - name: gnome-autoar
   sources:
   - type: git
diff --git a/build-aux/flatpak/org.gnome.NautilusMaster.yml b/build-aux/flatpak/org.gnome.NautilusMaster.yml
index 28c2dfb66..db0a5b2ed 100644
--- a/build-aux/flatpak/org.gnome.NautilusMaster.yml
+++ b/build-aux/flatpak/org.gnome.NautilusMaster.yml
@@ -89,6 +89,16 @@ modules:
   sources:
   - type: git
     url: https://gitlab.gnome.org/GNOME/tracker-miners.git
+- name: gnome-desktop
+  buildsystem: meson
+  build-options:
+    config-opts:
+    - "-Ddebug_tools=false"
+    - "-Ddesktop_docs=false"
+    - "-Dudev=disabled"
+  sources:
+  - type: git
+    url: https://gitlab.gnome.org/GNOME/gnome-desktop.git
 - name: gnome-autoar
   sources:
   - type: git
diff --git a/meson.build b/meson.build
index b7c7731dc..613006c52 100644
--- a/meson.build
+++ b/meson.build
@@ -66,30 +66,6 @@ add_project_arguments(
   language: 'c'
 )
 
-######################
-# Host system checks #
-######################
-
-host_system = host_machine.system()
-host_cpu = host_machine.cpu()
-unsupported_cpus = [
-  'alpha',
-  'ia64',
-  'm68k',
-  'parisc',
-  'parisc64',
-  'sh4',
-  'sparc',
-  'sparc64',
-]
-system_supports_seccomp = host_system == 'linux'
-cpu_supports_seccomp = not unsupported_cpus.contains(host_cpu)
-seccomp_required = system_supports_seccomp and cpu_supports_seccomp
-
-##########################
-# End host system checks #
-##########################
-
 ##################
 # Module imports #
 ##################
@@ -130,15 +106,8 @@ gio_unix = dependency('gio-unix-2.0', version: glib_ver)
 glib = dependency('glib-2.0', version: glib_ver)
 gmodule = dependency('gmodule-no-export-2.0', version: glib_ver)
 gnome_autoar = dependency('gnome-autoar-0', version: '>= 0.2.1')
-gsettings_desktop_schemas = dependency('gsettings-desktop-schemas')
+gnome_desktop = dependency('gnome-desktop-3.0', version: '>= 3.0.0')
 gtk = dependency('gtk+-3.0', version: '>= 3.22.27')
-if seccomp_required
-  message('seccomp required on this platform, make sure bubblewrap is available at runtime as well.')
-  seccomp = dependency('libseccomp')
-else
-  warning('The host does not support seccomp, thumbnailer sandboxing will be disabled. Such setups are not recommended, use at your own risk!')
-  seccomp = dependency('', required: false)
-endif
 selinux = []
 if get_option('selinux')
   selinux = dependency('libselinux', version: '>= 2.0')
@@ -147,8 +116,6 @@ tracker_sparql = dependency('tracker-sparql-2.0')
 x11 = dependency('x11')
 xml = dependency('libxml-2.0', version: '>= 2.7.8')
 
-fontconfig = dependency('fontconfig', required: false)
-
 ####################
 # End dependencies #
 ####################
@@ -167,15 +134,8 @@ endif
 
 application_id = 'org.gnome.Nautilus' + profile
 
-if fontconfig.found()
-  fontconfig_cache_path = fontconfig.get_pkgconfig_variable('cachedir')
-else
-  fontconfig_cache_path = join_paths(libdir, 'fontconfig/cache')
-endif
-
 conf.set_quoted('APPLICATION_ID', application_id)
 conf.set_quoted('GETTEXT_PACKAGE', 'nautilus')
-conf.set_quoted('INSTALL_PREFIX', prefix)
 conf.set_quoted('LOCALEDIR', join_paths(prefix, localedir))
 conf.set_quoted('NAME_SUFFIX', name_suffix)
 conf.set_quoted('NAUTILUS_DATADIR', join_paths(datadir, 'nautilus'))
@@ -183,14 +143,6 @@ conf.set_quoted('NAUTILUS_EXTENSIONDIR', join_paths(prefix, extensiondir))
 conf.set_quoted('PACKAGE_VERSION', meson.project_version())
 conf.set_quoted('PROFILE', profile)
 conf.set_quoted('VERSION', '@0@-@VCS_TAG@'.format(meson.project_version()))
-conf.set_quoted('FONTCONFIG_CACHE_PATH', fontconfig_cache_path)
-
-###################################################
-# gnome-desktop macros for thumbnailer sandboxing #
-###################################################
-
-conf.set('ENABLE_SECCOMP', seccomp.found())
-conf.set('HAVE_BWRAP', seccomp.found())
 
 conf.set('ENABLE_PACKAGEKIT', get_option('packagekit'))
 conf.set('ENABLE_PROFILING', get_option('profiling'))
diff --git a/src/meson.build b/src/meson.build
index 91c0a00c8..86d062161 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -53,10 +53,6 @@ libnautilus_sources = [
   'animation/ide-box-theatric.h',
   'animation/ide-cairo.c',
   'animation/ide-cairo.h',
-  'gnome-desktop/gnome-desktop-thumbnail.c',
-  'gnome-desktop/gnome-desktop-thumbnail.h',
-  'gnome-desktop/gnome-desktop-thumbnail-script.c',
-  'gnome-desktop/gnome-desktop-thumbnail-script.h',
   'gtk/nautilusgtkplacesview.c',
   'gtk/nautilusgtkplacesviewprivate.h',
   'gtk/nautilusgtkplacesviewrow.c',
@@ -277,10 +273,9 @@ nautilus_deps = [
   gio_unix,
   gmodule,
   gnome_autoar,
-  gsettings_desktop_schemas,
+  gnome_desktop,
   libgd_dep,
   nautilus_extension,
-  seccomp,
   selinux,
   tracker_sparql,
   xml
diff --git a/src/nautilus-properties-window.c b/src/nautilus-properties-window.c
index 464a131ca..9d9bd1c54 100644
--- a/src/nautilus-properties-window.c
+++ b/src/nautilus-properties-window.c
@@ -34,7 +34,7 @@
 #include <sys/stat.h>
 
 #define GNOME_DESKTOP_USE_UNSTABLE_API
-#include "gnome-desktop/gnome-desktop-thumbnail.h"
+#include <libgnome-desktop/gnome-desktop-thumbnail.h>
 
 #include "nautilus-enums.h"
 #include "nautilus-error-reporting.h"
diff --git a/src/nautilus-thumbnails.c b/src/nautilus-thumbnails.c
index b90dc0daa..991bfb6f6 100644
--- a/src/nautilus-thumbnails.c
+++ b/src/nautilus-thumbnails.c
@@ -40,7 +40,7 @@
 #include <sys/wait.h>
 #include <unistd.h>
 #include <signal.h>
-#include "gnome-desktop/gnome-desktop-thumbnail.h"
+#include <libgnome-desktop/gnome-desktop-thumbnail.h>
 #define DEBUG_FLAG NAUTILUS_DEBUG_THUMBNAILS
 #include "nautilus-debug.h"
 
-- 
2.21.0

